lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -254,6 +255,23 @@ def resource_output_paths(resource_input_paths)
               File.join(base_path, File.basename(basename, extname) + output_extension)
             end.uniq
           end
+
+          # Returns the framework output paths for the given input paths
+          #
+          # @param  [Array<Target::FrameworkPaths>] framework_input_paths
+          #         The framework input paths to map to output paths.
+          #
+          # @return [Array<String>] The framework output paths
+          #
+          def framework_output_paths(framework_input_paths)
```

Comment:
alternatively we can have each instance of `FrameworkPaths` declare a `def output_path` and then `map(&:output_path)`?

Reply:
I think the logic for where the frameworks are placed in the built product shouldn't live within that model. it's a bit weird though because the embed frameworks script actually duplicates the logic of both the input and the output paths

https://github.com/CocoaPods/CocoaPods/blob/92385cabea6b572867ab39ec742c862cd942fc07/lib/cocoapods/generator/embed_frameworks_script.rb#L76-L84

so really, this is only controlling what the script's input and output paths are set in Xcode, and it doesn't affect the actual script contents at all. should probably unify that logic too

Reply:
this also matches nicely with the `resource_output_paths` function right above this one, so at least its consistent? 🤷‍♂️